### PR TITLE
Let SEP-31 POST transaction depend on SEP-38 POST quote

### PR DESCRIPTION
**Pull request by erika-sdf**
Specify quote_id in transaction POST request.

### CODE REVIEW

File: @stellar/anchor-tests/src/tests/sep31/transactions.ts [hunk 1]
```diff
@@ -55,13 +56,15 @@ const canCreateTransaction: Test = {
     hasDirectPaymentServer,
     hasExpectedAssetEnabled,
     differentMemosSameAccount,
+    canCreateQuote,
```
Comment (marcelosalloum) [inline @stellar/anchor-tests/src/tests/sep31/transactions.ts:59]: The code looks great but there are a few edge cases that make your change tricky and not safe for deployment:
1. There are Anchors that only implement SEP-31 without implementing SEP-38. In those cases, the dependency `canCreateQuote` will fail, as well as this whole sep31 test. You'll need to create a new set of tests that covers anchors implementing both SEP-31+SEP-38. You should create a new file for these tests and decide to add it to the list of tests here:
https://github.com/stellar/stellar-anchor-tests/blob/11bf82c295d01ffea018cf6e3edc048a3156a610/@stellar/anchor-tests/src/helpers/test.ts#L346-L351
2. Additionally, if you want to support anchors with `quotes_required: true`, you should update the `sep31/transactions#canCreateTransaction` test by checking if they are required, and if that's the case, make the same request as the code was doing before and validate if the resulting status code is `400`.

@JakeUrban: do you have any other input here?

**Review by marcelosalloum — COMMENTED** [review]
The code looks great but there are a few edge cases that make your change tricky and not safe for deployment. I'm explaining that in the comment below:

File: @stellar/anchor-tests/src/tests/sep31/transactions.ts (hunk 1 above)
Comment (erika-sdf) [reply @stellar/anchor-tests/src/tests/sep31/transactions.ts]: I've made the following changes:

* Added a set of SEP31+SEP38 tests that execute when `quotes_supported: true`
* Added a check for `quotes_required` in SEP31 tests. If it is false, `canCreateTransaction` and it's dependencies if are executed. If `quotes_required: true`, `canCreateTransaction` expects a 400, and its dependencies don't check for anything.

File: @stellar/anchor-tests/src/tests/sep31/transactions.ts [hunk 2]
```diff
@@ -117,7 +121,25 @@ const canCreateTransaction: Test = {
...
+      this.context.expects.sep31InfoObj,
+      assetCode,
+    );
+
+    // If quotes are required, ignore this test, this will be addressed in SEP 31+38 tests
+    if (quotesRequired) {
+      await makeRequest(postTransactionsCall, 400, result, "application/json");
+      this.context.provides.transactionId = null;
+      return result;
+    }
```
Comment (JakeUrban) [inline @stellar/anchor-tests/src/tests/sep31/transactions.ts:139]: So this is testing that if quotes are required, that the `POST /transactions` call fails with a 400? Maybe we should add a failure mode that describes this, that way its clear to the user why the test failed.

File: @stellar/anchor-tests/src/tests/sep31And38/transactions.ts [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    );
+
+    // If quotes are not supported, ignore this test, this will be addressed in SEP 31 tests
+    if (!quotesSupported) {
+      this.context.provides.transactionId = null;
+      return result;
+    }
+
+    const splitAsset =
+      this.context.expects.sep38QuoteResponseObj.sell_asset.split(":", 3);
```
Comment (JakeUrban) [inline @stellar/anchor-tests/src/tests/sep31And38/transactions.ts:90]: Are we verifying the format of this is correct before using it?

**Review by JakeUrban — COMMENTED** [review]
This looks great, we've been needing these tests for quite a bit now. Left a couple comments but nothing blocking.

File: @stellar/anchor-tests/src/tests/sep31And38/transactions.ts (hunk 3 above)
Comment (erika-sdf) [reply @stellar/anchor-tests/src/tests/sep31And38/transactions.ts:90]: Yes, this has been tested.

File: @stellar/anchor-tests/src/tests/sep31/transactions.ts (hunk 2 above)
Comment (erika-sdf) [reply @stellar/anchor-tests/src/tests/sep31/transactions.ts:139]: I see what you're saying, but I think we also don't want this test to fail when it's being used for validation if a 400 is the expected behavior.

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Comment by marcelosalloum** [comment]
Please don't forget to prepare for a new version shipment by:
1. Updating the project version, like in https://github.com/stellar/stellar-anchor-tests/pull/85/files
2. Updating the Changelog, like in https://github.com/stellar/stellar-anchor-tests/pull/80/files

You can do these changes in this same pull request, so you can ship things faster.

After you do that and merge this PR, you just need to create a [new release](https://github.com/stellar/stellar-anchor-tests/releases) with the. same version number you used in package.json and the same text you used in the changelog.
